runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3664,6 +3664,151 @@ TR::Register *J9::X86::TreeEvaluator::longBitCount(TR::Node *node, TR::CodeGener
+       generateX86MemoryReference(temp2Reg, offsetof(J9ROMClass, modifiers), cg), J9AccInterface, cg);
+   generateLabelInstruction(JE4, node, isClassLabel, cg);
+   
+   // Obtain I-Table
+   generateRegMemInstruction(LRegMem(), node, temp1Reg, generateX86MemoryReference(temp3Reg, offsetof(J9Class, iTable), cg), cg);
+   // Loop through I-Table
+   generateLabelInstruction(LABEL, node, iTableLoopLabel, cg);
+   generateRegRegInstruction(TESTRegReg(), node, temp1Reg, temp1Reg, cg);
+   generateLabelInstruction(JE4, node, throwLabel, cg);
+   auto interfaceMR = generateX86MemoryReference(temp1Reg, offsetof(J9ITable, interfaceClass), cg);
+   generateMemRegInstruction(CMPMemReg(), node, interfaceMR, castClassReg, cg);
+   generateRegMemInstruction(LRegMem(), node, temp1Reg, generateX86MemoryReference(temp1Reg, offsetof(J9ITable, next), cg), cg);
+   generateLabelInstruction(JNE4, node, iTableLoopLabel, cg);
+   
+   // Found from I-Table
+   generateLabelInstruction(JMP4, node, fallThruLabel, cg);
+   
+   // cast class is non-interface class
+   generateLabelInstruction(LABEL, node, isClassLabel, cg);
+   // equality test
+   generateRegRegInstruction(CMPRegReg(use64BitClasses), node, temp3Reg, castClassReg, cg);
+   generateLabelInstruction(JE4, node, fallThruLabel, cg);
+   
+   // class not equal,
+   // if cast class depth >= min depth, compare with object class depth
+   // store cast class depth in temp2 for later use
+   // class depth mask must be low 16 bits to safely load without the mask.
+   static_assert(J9AccClassDepthMask == 0xffff, "J9_JAVA_CLASS_DEPTH_MASK must be 0xffff");
+   generateRegMemInstruction(TR::Compiler->target.is64Bit()? MOVZXReg8Mem2 : MOVZXReg4Mem2, node,
+            temp2Reg, generateX86MemoryReference(castClassReg, offsetof(J9Class, classDepthAndFlags), cg), cg);
+   generateRegImmInstruction(CMP2RegImm2, node, temp2Reg, ((int16_t)(cg->comp()->getOptions()->_minimumSuperclassArraySize)), cg);
+   generateLabelInstruction(JB4, node, checkSuperClassArrayLabel, cg);
+   
+   // cast class depth >= obj class depth, fail
+   generateRegMemInstruction(CMP2RegMem, node, temp2Reg, generateX86MemoryReference(temp3Reg, offsetof(J9Class, classDepthAndFlags), cg), cg);
+   generateLabelInstruction(JAE4, node, throwLabel, cg);
+   
+   // check obj class's super class array entry
+   generateLabelInstruction(LABEL, node, checkSuperClassArrayLabel, cg);
+   generateRegMemInstruction(LRegMem(), node, temp1Reg, generateX86MemoryReference(temp3Reg, offsetof(J9Class, superclasses), cg), cg);
+   generateRegMemInstruction(CMPRegMem(use64BitClasses), node, castClassReg,
+       generateX86MemoryReference(temp1Reg, temp2Reg, TR::Compiler->target.is64Bit()?3:2, cg), cg);
+   generateLabelInstruction(JNE4, node, throwLabel, cg);
+   
+   // throw classCastException
+   {
+      TR_OutlinedInstructionsGenerator og(throwLabel, node, cg);
+      //generateLabelInstruction(LABEL, node, throwLabel, cg);
```

Comment:
This comment seems unnecessary, I'd prefer removing it.

Reply:
Addressed.